iceoryx_posh/test/moduletests/test_roudi_iceoyx_roudi_memory_manager.cpp

```diff
@@ -0,0 +1,151 @@
+{
+    auto testResult = m_roudiMemoryManagerTest->portPool();
+    EXPECT_THAT(testResult, Eq(iox::cxx::nullopt_t()));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, CreateAndAnnouceMemoryHasNoError)
+{
+    auto testResult = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+    EXPECT_THAT(testResult.has_error(), Eq(false));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, MgmtMemoryProviderReturnNonNullPtr)
+{
+    auto testResult = m_roudiMemoryManagerTest->mgmtMemoryProvider();
+    EXPECT_THAT(testResult, Not(Eq(nullptr)));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, AcquiringIntrospectionMemoryManagerAfterCreateAndAnnounceMemoryIsSuccessful)
+{
+    auto tr = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    EXPECT_THAT(tr.has_error(), Eq(false));
+
+    auto result = m_roudiMemoryManagerTest->introspectionMemoryManager();
+    EXPECT_THAT(result, Not(Eq(iox::cxx::nullopt_t())));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, AcquiringSegmentManagerAfterCreateAndAnnounceMemoryIsSuccessful)
+{
+    auto tr = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    EXPECT_THAT(tr.has_error(), Eq(false));
+
+    auto resultTest = m_roudiMemoryManagerTest->segmentManager();
+    EXPECT_THAT(resultTest, Not(Eq(iox::cxx::nullopt_t())));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, AcquiringPortPoolAfterCreateAndAnnounceMemoryIsSuccessful)
+{
+    auto tr = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    EXPECT_THAT(tr.has_error(), Eq(false));
+
+    auto testResult = m_roudiMemoryManagerTest->portPool();
+    EXPECT_THAT(testResult, Not(Eq(iox::cxx::nullopt_t())));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, DestroyMemoryReturnNoError)
+{
+    auto testResult = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    auto result = m_roudiMemoryManagerTest->destroyMemory();
+
+    EXPECT_THAT(result.has_error(), Eq(false));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, DestroyMemoryIntrospectionMemoryManagerReturnNullOpt)
+{
+    auto testResult = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    auto result = m_roudiMemoryManagerTest->destroyMemory();
+
+    auto res = m_roudiMemoryManagerTest->introspectionMemoryManager();
+    EXPECT_THAT(res, Eq(iox::cxx::nullopt_t()));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, DestroyMemorySegmentManagerReturnNullOpt)
+{
+    auto testResult = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    auto result = m_roudiMemoryManagerTest->destroyMemory();
+
+    auto resultTest = m_roudiMemoryManagerTest->segmentManager();
+    EXPECT_THAT(resultTest, Eq(iox::cxx::nullopt_t()));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, CreateAndAnnouceMemoryFailingAfterCalledTwoTimes)
+{
+    auto testResult = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    auto result = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    auto res = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    EXPECT_THAT(res.has_error(), Eq(true));
```

Comment:
please check that the second call to `createAndAnnounceMemory` also results in an error and also check for the specific error value

Reply:
Done

Reply:
Did you push your changes?

Reply:
My bad, i thought It was pushed. It is now :)

Reply:
no problem :)
Can you please also create the test where `destroyMemory` is called two times without error